Cython/Compiler/ModuleNode.py

```diff
@@ -2837,12 +3083,19 @@ def generate_pymoduledef_struct(self, env, code):
         code.putln("#endif")
 
         code.putln("")
+        code.putln('#ifdef __cplusplus')
+        code.putln('namespace {')
+        code.putln("struct PyModuleDef %s = {" % Naming.pymoduledef_cname)
+        code.putln('#else')
         code.putln("static struct PyModuleDef %s = {" % Naming.pymoduledef_cname)
```

Comment:
`static` or not?

Reply:
C++ is not compatible with the C way of forward declaring a static struct. The C standard specifies that within a translation unit, the declaration of an identifier with internal linkage belongs to the same object. C++ does not specify that, therefore, causing a compile error. The correct way to model this in C++ is by forward declaring using the anonymous namespace and extern along with another declaration setting the memory.

I will mark this as resolved but feel free to push back with more questions!